crates/core/tedge_api/src/message_log.rs

```diff
@@ -68,15 +68,22 @@ pub struct MessageLogWriter {
 }
 
 impl MessageLogWriter {
-    pub fn new<P>(log_dir: P) -> Result<MessageLogWriter, std::io::Error>
+    pub fn new<P>(log_dir: P, truncate: bool) -> Result<MessageLogWriter, std::io::Error>
```

Comment:
Instead of adding a `bool` to the constructor, it would be better to add a new constructor.

```
    pub fn new<P>(log_dir: P) -> Result<MessageLogWriter, std::io::Error> {
             // unchanged.
    }

    pub fn new_truncated<P>(log_dir: P) -> Result<MessageLogWriter, std::io::Error> {
             // truncating the file first.
    }

```